Blender-VF-Exporter/Blender-VF-Exporter.py

```diff
@@ -0,0 +1,435 @@
+						export_frame_range=True,
+						export_frame_step=1,
+						export_force_sampling=True,
+						export_nla_strips=True,
+						export_def_bones=True,
+						export_optimize_animation_size=True,
+						export_current_frame=False,
+						export_skins=True,
+						export_all_influences=False,
+
+						export_morph=True,
+						export_morph_normal=True,
+						export_morph_tangent=False,
+
+						export_lights=False,
+						will_save_settings=False,
+						filter_glob='*.glb;*.gltf')
+
+				if combined:
+					break
+
+			for i in range(undo_steps):
+				bpy.ops.ed.undo()
+		elif format == "STL":
+			batch = 'OFF' if combined else 'OBJECT'
+			output = location + file_name + file_format if combined else location
+			bpy.ops.export_mesh.stl(
+				filepath=output,
+				ascii=False,
+				check_existing=False, # Dangerous!
+				use_selection=True,
+				batch_mode=batch,
+
+				global_scale=1.0,
+				use_scene_unit=False,
+				use_mesh_modifiers=True,
+
+				axis_forward='Y',
+				axis_up='Z',
+				filter_glob='*.stl')
+
+		elif format == "CSV":
+
+			frame_current = bpy.context.scene.frame_current
+
+			frame_start = bpy.context.scene.frame_start
+			frame_end = bpy.context.scene.frame_end
+			space = bpy.context.scene.vf_delivery_settings.csv_position
+
+			for obj in bpy.context.selected_objects:
+
+				array = [["x","y","z"]]
+				for i in range(frame_start, frame_end + 1):
+					bpy.context.scene.frame_set(i)
+					loc, rot, scale = obj.matrix_world.decompose() if space == "WORLD" else obj.matrix_local.decompose()
+					array.append([loc.x, loc.y, loc.z])
+
+				np.savetxt(
+					location + obj.name + file_format,
+					array,
+					delimiter =",",
+					newline='\n',
+					fmt ='% s'
+					)
+
+			bpy.context.scene.frame_set(frame_current)
+
+		if format != "CSV":
+			bpy.ops.ed.undo()
+
+		bpy.ops.object.mode_set(mode=mode)
+		return {'FINISHED'}
+
+
+class vfDeliveryPreferences(bpy.types.AddonPreferences):
+	bl_idname = __name__
+
+
+	enable_uvmap_experimental: bpy.props.BoolProperty(
+		name='Enable experimental UVMap conversion',
+		description='Attempts to convert the first named attribute output from a Geometry Nodes modifier into a UV map recognisable by file exporters',
+		default=False)
+
+
+	def draw(self, context):
+		layout = self.layout
+		layout.prop(self, "enable_uvmap_experimental")
+
+class vfDeliverySettings(bpy.types.PropertyGroup):
+	file_type: bpy.props.EnumProperty(
+		name='Pipeline',
+		description='Sets the format for delivery output',
+		items=[
+			('ABC', 'ABC — Static', 'Export Alembic binary file from frame 0'),
+			('FBX', 'FBX — Unity3D', 'Export FBX binary file for Unity'),
+			('GLB', 'GLB — ThreeJS', 'Export GLTF compressed binary file for ThreeJS'),
+			('STL', 'STL — 3D Printing', 'Export individual STL file of each selected object for 3D printing'),
+			('CSV', 'CSV — Position', 'Export CSV file of the selected object\'s position for all frames within the render range')
+			],
+		default='STL')
```

Comment:
Please set the default to FBX
Do you develop a package for 3D printing!!?
It is true that we added the rest of the formats, but our main target is unity engine